code/Forms/UnsavedChangesIndicator.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace SilverStripe\Admin\Forms;
+
+use RuntimeException;
+use SilverStripe\Forms\FormAction;
+
+/**
+ * A visual indicator to show whether a form has unsaved changes.
+ *
+ * Note that this isn't strictly a FormAction as it doesn't perform any action when clicked.
+ * However, extending FormAction allows us to easily integrate with the existing form action
+ * architecture (e.g. placement in the form actions area).
+ */
+class UnsavedChangesIndicator extends FormAction
+{
+    private static array $minutes = [
+        'notice' => 5,
+        'warning' => 10,
+    ];
+
+    protected $schemaComponent = 'UnsavedChangesIndicator';
+
+    public function getSchemaDataDefaults()
+    {
+        // This is used for initialising the React component
+        // Refer to silverstrpie/admin UnsavedChangesIndicator.js
+        $data = parent::getSchemaDataDefaults();
+        $data['minutes'] = $this->getMinutes();
+        return $data;
+    }
+
+    public function getAttributes()
+    {
+        // This is used for entwine initialisation
+        // Refer to silverstrpie/admin UnsavedChangesIndicatorEntwine.js
+        $attributes = parent::getAttributes();
+        $attributes['data-minutes'] = json_encode($this->getMinutes());
+        return $attributes;
+    }
+
+    private function getMinutes(): array
+    {
+        $minutes = UnsavedChangesIndicator::config()->get('minutes');
+        $this->validateConfig($minutes);
+        return $minutes;
+    }
+
+    private function validateConfig(array $minutes): void
+    {
+        if (!isset($minutes['notice']) || !isset($minutes['warning'])) {
+            throw new RuntimeException('Config must contain both "notice" and "warning" keys');
+        }
+        // Allow decimals here for partial minutes, as this is required for behat
+        // testing and develoment so that we don't have to wait long periods of time
+        if (!is_numeric($minutes['notice']) || !is_numeric($minutes['warning'])) {
+            throw new RuntimeException('Config values must be numeric');
+        }
+        if ($minutes['notice'] < 0 || $minutes['warning'] < 0) {
+            throw new RuntimeException('Config values must be positive or zero');
+        }
+        if ($minutes['warning'] !== 0 && $minutes['warning'] < $minutes['notice']) {
+            throw new RuntimeException('Config "warning" value must be greater than or equal to "notice" value');
```

Comment:
```suggestion
            throw new RuntimeException('minutes config must contain both "notice" and "warning" keys');
        }
        // Allow decimals here for partial minutes, as this is required for behat
        // testing and develoment so that we don't have to wait long periods of time
        if (!is_numeric($minutes['notice']) || !is_numeric($minutes['warning'])) {
            throw new RuntimeException('minutes config values must be numeric');
        }
        if ($minutes['notice'] < 0 || $minutes['warning'] < 0) {
            throw new RuntimeException('minutes config values must be positive or zero');
        }
        if ($minutes['warning'] !== 0 && $minutes['warning'] < $minutes['notice']) {
            throw new RuntimeException('minutes config "warning" value must be greater than or equal to "notice" value');
```
Just providing a little more context in the exception messages to make it clear _which_ config is incorrect. 

Yes this class only has one direct config currently, but someone who's in a rush and doesn't look at where the exception is being thrown might get frustrated because "which config? This site has a billion config parameters"

Reply:
Updated